fbpic/main.py

```diff
@@ -315,44 +319,46 @@ def step(self, N=1, correct_currents=True,
             if show_progress and self.comm.rank==0:
                 progression_bar( i_step, N, measured_start )
 
-            # Exchanges to prepare for this iteration
-            # ---------------------------------------
-
             # Run the diagnostics
-            # (E, B, rho are defined at time n)
-            # (J, x, p are defined at time n+1/2)
+            # (E, B, rho, x are defined at time n; J, p at time n-1/2)
             for diag in self.diags:
-                # Check if the fields should be written at
-                # this iteration and do it if needed.
-                # (Send the data to the GPU if needed.)
+                # Check if the diagnostic should be written at this iteration
+                # and write it, if it is the case.
+                # (If needed: bring rho/J from spectral space, where they
+                # were smoothed/corrected, and copy the data from the GPU.)
```

Comment:
Maybe we should add a note here, that says that E, B, rho, x are defined at n and J,p are defined at n-1/2